arrow/src/main/java/org/apache/iceberg/arrow/vectorized/VectorizedArrowReader.java

```diff
@@ -332,6 +334,10 @@ public static VectorizedArrowReader nulls() {
     return NullVectorReader.INSTANCE;
   }
 
+  public static VectorizedArrowReader positions() {
+    return PositionVectorReader.INSTANCE;
+  }
```

Comment:
I am unsure if returning a singleton instance for `PostitionVectorReader` is safe, since it contains a member variable `rowStart` which seems to differ for every row group created. Can there be a possibility of multiple tasks running on the same executor JVM and referring to different `PostionVectorReaders` at the same time?

Reply:
Thanks for your comment.

The `INSTANCE` is defined to return a new `PosistionVectorReader`. It doesn't have a class scope field such as `instance` and the null checking logic, so it is not a singleton.

IIUC, spark will assign the number of `spark.executor.cores` of tasks per executor.

Reply:
The `INSTANCE` field is defined is static, so I am guessing `new PositionVectorReader()` will only be called once when the class is being loaded by the JVM. So this seems like mimicking a singleton to me, unless I am reading the things wrongly.

Reply:
You are right! I forgot the `static` keyword...